src/GraphQL/Types/Scalars/DateTimeGraphType.cs

```diff
@@ -20,13 +21,39 @@ public DateTimeGraphType()
         }
 
         /// <inheritdoc/>
-        public override object ParseLiteral(IValue value)
-            => value is StringValue stringValue ? ParseValue(stringValue.Value) : null;
+        public override object ParseLiteral(IValue value) => value switch
+        {
+            StringValue stringValue => ParseDate(stringValue.Value),
+            NullValue _ => null,
+            _ => ThrowLiteralConversionError(value)
+        };
 
         /// <inheritdoc/>
-        public override object ParseValue(object value) => ValueConverter.ConvertTo(value, typeof(DateTime));
+        public override object ParseValue(object value) => value switch
+        {
+            DateTime _ => value,
+            string s => ParseDate(s),
+            null => null,
+            _ => ThrowValueConversionError(value)
+        };
+
+        private static DateTime ParseDate(string stringValue)
+        {
+            // ISO-8601 format
+            if (DateTime.TryParseExact(stringValue, "yyyy'-'MM'-'dd'T'HH':'mm':'ss.FFFFFFFK", DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeUniversal | DateTimeStyles.AdjustToUniversal, out var date))
+            {
+                return date;
+            }
+
+            throw new FormatException($"Could not parse date. Expected ISO-8601 format. Value: {stringValue}");
+        }
 
         /// <inheritdoc/>
-        public override IValue ToAST(object value) => new StringValue(((DateTime)value).ToString("O")); // "O" is the proper ISO 8601 format required
+        public override object Serialize(object value) => value switch
+        {
+            DateTime d => d.ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss.FFFFFFFK", DateTimeFormatInfo.InvariantInfo), // ISO-8601 format (without unnecessary decimal places, allowed by ISO-8601)
```

Comment:
The ISO-8601 serialization format is now specified in code, rather than relying on System.Text.Json to provide the conversion.  The format has not changed and tests that rely on this exact format still pass.  Note that the "O" format is similar but always prints the fractional parts of the second, which is not required by ISO-8601.

Reply:
Please add comments here about differences with 'O' format.

Reply:
Done